lstchain/high_level/interpolate.py

```diff
@@ -0,0 +1,577 @@
+        tri = Delaunay(mc_params_sel)
+    except ValueError:
+        print("Not enough grid values for Delaunay triangulation")
+        return new_irfs
+
+    target_in_simplex = tri.find_simplex(data_val_sel)
+
+    if not use_nearest_irf_node:
+        if target_in_simplex == -1:
+            # The target values are not contained in any Delaunay triangle formed
+            # by the paramters of the list of IRFs provided.
+            # So just include the IRF with the closest parameter values
+            # to the target values
+
+            index = distance.cdist([data_val_sel], mc_params_sel).argmin()
+            print("Target value is outside interpolation. Using the nearest IRF.")
+            index = get_nearest_az_node(
+                mc_params_sel, index, mc_params_full, data_val_full
+            )
+            new_irfs.append(irfs[index])
+        else:
+            # Just select the IRFs that are needed for the Delaunay triangulation
+            for i in tri.simplices[target_in_simplex]:
+                i_sel = get_nearest_az_node(
+                    mc_params_sel, i, mc_params_full, data_val_full
+                )
+                new_irfs.append(irfs[i_sel])
+    else:
+        index = distance.cdist([data_val_sel], mc_params_sel).argmin()
+        print("Using the nearest IRF.")
+        index = get_nearest_az_node(mc_params_sel, index, mc_params_full, data_val_full)
+        new_irfs.append(irfs[index])
+
+    return new_irfs
+
+
+def get_nearest_az_node(irf_params_sel, index, irf_params_full, target_params_full):
+    """
+    Check to see if a given IRF node overlaps with another node, in the
+    interpolation parameter space, and to select based on the azimuth (hard-
+    coded to be index 2 for now) angle, the nearest node to the target.
+    """
+    # Remove the numerical variations by shortening the precision of values
+    irf_params_short = np.around(irf_params_sel, 3)
+
+    # Find the uniques set of nodes, to compare and choose the indices
+    irf_unique, irf_num = np.unique(irf_params_short, axis=0, return_counts=True)
+    idx = np.flatnonzero((irf_unique == irf_params_short[index]).all(1))
+
+    if irf_num[idx] > 1:
+        # Only using the overlapping nodes
+        idx_list = np.flatnonzero((irf_params_short == irf_params_short[index]).all(1))
+        # Finding the shortest distance with respect to target azimuth
+        diff = np.abs(irf_params_full[idx_list, 2] - target_params_full[2])
+        idx = idx_list[np.where(diff == diff.min())[0]][0]
+    else:
+        # if there are no overlapping nodes
+        idx = index
+    return idx
+
+
+def compare_irfs(irfs):
+    """
+    Compare the given list of IRFs with data binning and relevant
+    metadata values.
+
+    Parameters
+    ----------
+    irfs: List of IRFs to compare
+        'List'
+    Returns
+    -------
+    : Boolean indicating whether the given list of IRFs are comparable
+    """
+    bin_bool = False
+    meta_bool = True
```

Comment:
Is there a particular reason you default one to True and the other to False? 
I would do both as False if not.
(and thus change != to == l215)

Reply:
Here, I went with the idea of making the final boolean be `False` whenever the code sees different metadata values, and so, not do anything when the values match.
I agree, that this function may not be using the best practice of playing with python booleans, but at the time, I could not get a simpler approach.

Reply:
You are right i missed the loop in the logic. It is good then (also for the next comment)